src/main/java/org/jahia/modules/sam/healthcheck/probes/ModulesSpringUsageProbe.java

```diff
@@ -0,0 +1,170 @@
+                    LOGGER.debug("Ignoring Jahia provided module: {}", bundle.getSymbolicName());
+                    continue;
+                }
+
+                LOGGER.debug("Exploring bundle: {}", bundle.getSymbolicName());
+                // Explore using ApplicationContextConfiguration (presence of spring context xml file)
+                ApplicationContextConfiguration config = new ApplicationContextConfiguration(bundle);
+                if (config.isSpringPoweredBundle()) {
+                    LOGGER.info("Detected Spring powered module: {}", bundle.getSymbolicName());
+                    springUsages.add(new SpringUsageInfo(bundle, "detected a Spring context xml file"));
+                }
+                // Explore OSGI imports for Spring Framework related package or Jahia Spring Related package
+                String imports = bundle.getHeaders().get("Import-Package");
+                if (imports != null) {
+                    for (String importStatement : imports.split("(?<=\\S),(?!\\h*\\d)\\h*")) {
+                        LOGGER.debug("Exploring import package: {}", importStatement);
+                        if (importStatement.startsWith("org.springframework")) {
+                            LOGGER.info("Spring related package: {} imported in module: {}", importStatement, bundle.getSymbolicName());
+                            springUsages.add(new SpringUsageInfo(bundle,
+                                    "detected a Spring related package imported in OSGI manifest: [" + importStatement + "]"));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    protected static class SpringUsageInfo {
+        private String module;
+        private String usage;
+
+        public SpringUsageInfo(Bundle bundle, String usage) {
+            this.module = "(id:".concat(String.valueOf(bundle.getBundleId())).concat(") ").concat(bundle.getSymbolicName())
+                    .concat(" - ").concat(bundle.getVersion().toString());
+            this.usage = usage;
+        }
+
+        public String getModule() {
+            return module;
+        }
+
+        public void setModule(String module) {
+            this.module = module;
+        }
+
+        public String getUsage() {
+            return usage;
+        }
+
+        public void setUsage(String usage) {
+            this.usage = usage;
+        }
+
+        @Override public String toString() {
+            return "module[" + module + "] " + usage + "   ";
+        }
+    }
+
+    private String checksum(Bundle[] bundles){
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            Arrays.stream(bundles).map(Bundle::getLastModified).forEach(lastModified -> md.update(String.valueOf(lastModified).getBytes()));
+            return Base64.getEncoder().encodeToString(md.digest());
+        } catch (Exception ignored) {
+        }
+        return "";
+    }
```

Comment:
We could also implement a simple OSGI bundle listener, and every time it's triggered invalidate previous results.
Actually this could be very useful option, as it could also benefit for the ModuleBundleVersions.
We should consider adding this accross the multiple probes depending on bundle changes.